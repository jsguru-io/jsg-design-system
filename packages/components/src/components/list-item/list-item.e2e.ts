/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newE2EPage } from '@stencil/core/testing';

describe('scale-list-item', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-list-item>default</scale-list-item>');
    const element = await page.find('scale-list-item');
    expect(element).toHaveClass('hydrated');
  });
});
