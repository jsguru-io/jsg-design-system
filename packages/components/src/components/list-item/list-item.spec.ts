/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newSpecPage } from '@stencil/core/testing';
import { ListItem } from './list-item';

describe('ListItem', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [ListItem],
      html: `<scale-list-item>default</scale-list-item>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
