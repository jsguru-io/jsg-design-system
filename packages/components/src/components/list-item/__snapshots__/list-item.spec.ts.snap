// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<scale-list-item role="listitem">
  <mock:shadow-root>
    <div class="list-item list-item--unordered" part="base unordered">
      <slot></slot>
      <div class="list-item__nested-list" hidden="" part="nested-list">
        <slot name="nested"></slot>
      </div>
    </div>
  </mock:shadow-root>
  default
</scale-list-item>
`;
